src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridApiTransport.php

```diff
@@ -190,6 +191,8 @@ private function getAttachments(Email $email): array
 
     private function getEndpoint(): ?string
     {
-        return ($this->host ?: self::HOST).($this->port ? ':'.$this->port : '');
+        $host = $this->host ?: str_replace('%region_dot%', 'eu' !== ($this->region ?: 'eu') ? $this->region.'.' : '', self::HOST);
```

Comment:
I suggest to replace the nested ternary expressions with `if` statements. This will increase readability and probably also reveal why the tests are failing.